packages/ckeditor5-enter/src/entercommand.ts

```diff
@@ -29,75 +29,89 @@ export default class EnterCommand extends Command {
 	 * @inheritDoc
 	 */
 	public override execute(): void {
-		const model = this.editor.model;
-		const doc = model.document;
-
-		model.change( writer => {
-			enterBlock( this.editor.model, writer, doc.selection, model.schema );
+		this.editor.model.change( writer => {
+			this.enterBlock( writer );
 			this.fire<EnterCommandAfterExecuteEvent>( 'afterExecute', { writer } );
 		} );
 	}
-}
 
-export type EnterCommandAfterExecuteEvent = {
-	name: 'afterExecute';
-	args: [ { writer: Writer } ];
-};
+	/**
+	 * Splits a block where the document selection is placed, in the way how the <kbd>Enter</kbd> key is expected to work:
+	 *
+	 *		<p>Foo[]bar</p>   ->   <p>Foo</p><p>[]bar</p>
+	 *		<p>Foobar[]</p>   ->   <p>Foobar</p><p>[]</p>
+	 *		<p>Fo[ob]ar</p>   ->   <p>Fo</p><p>[]ar</p>
+	 *
+	 * In some cases, the split will not happen:
+	 *
+	 * 1. The selection parent element is a limit element (`<figcaption>A[bc]d</figcaption>` -> `<figcaption>A[]d</figcaption>`).
+	 * 2. The selection spans over multiple elements (`<h>x[x</h><p>y]y<p>` -> `<h>x</h><p>[]y</p>`).
+	 *
+	 * @param {module:engine/model/writer~Writer} writer Writer to use when performing the enter action.
+	 * @returns {Boolean} `true` if a block was split, `false` otherwise.
+	 */
+	public enterBlock( writer: Writer ): boolean {
```

Comment:
If you wan't it to be visible in API Docs, add it to `_src/entercommand.js`. However, it must be JsDoc compatible (so include `{module:engine/model/writer~Writer}` and `{Boolean}` as you did above). You needn't bother with `execute()`, though.